### PR TITLE
Add test for corruption and repair and verification

### PR DESCRIPTION
**Pull request by neunhoef**
This is a sub-PR to add more tests.

### CODE REVIEW

File: tests/js/server/shell/revision-trees-corrupt-repair-noncluster.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      }
+    }
+    console.log("Still waiting for pending updates in tree: ", pending);
+    wait(0.5);
+  }
+}
+
+function corruptRepairSuite () {
+'use strict';
+jsunity.jsUnity.attachAssertions();
```
Comment (jsteemann) [inline tests/js/server/shell/revision-trees-corrupt-repair-noncluster.js:55]:
```suggestion
  'use strict';
  jsunity.jsUnity.attachAssertions();
```

File: arangod/RocksDBEngine/RocksDBV8Functions.cpp [hunk 2]
```diff
@@ -297,6 +379,14 @@ void RocksDBV8Functions::registerResources() {
...
                        TRI_V8_ASCII_STRING(isolate, "_revisionTreeCorrupt"),
                        JS_CollectionRevisionTreeCorrupt);
+  // get trees from RAM and freshly computed
+  TRI_AddMethodVocbase(isolate, rt,
+                       TRI_V8_ASCII_STRING(isolate, "_revisionTreeVerification"),
+                       JS_CollectionRevisionTreeVerification);
+  // rebuildRevisionTree
+  TRI_AddMethodVocbase(isolate, rt,
+                       TRI_V8_ASCII_STRING(isolate, "_revisionTreeRebuild"),
+                       JS_CollectionRevisionTreeRebuild);
```
[the remark this answers is not in the document]
Comment (neunhoef) [reply arangod/RocksDBEngine/RocksDBV8Functions.cpp:389]: These are just the server functions, to which we do not have access anyway. Therefore, I would leave it out of the production code, but I really do not care.

File: js/client/modules/@arangodb/arango-collection.js [hunk 3]
```diff
@@ -1527,3 +1527,30 @@ ArangoCollection.prototype.loadIndexesIntoMemory = function () {
...
+/// @brief MerkleTreeVerification
+//////////////////////////////////////////////////////////////////////////////
+
+ArangoCollection.prototype._revisionTreeVerification = function() {
+  var batch = this._database._connection.POST(this._prefixurl('/_api/replication/batch'), {ttl : 3600});
+  if (!batch.hasOwnProperty("id")) {
+    throw "Could not create batch!";
+  }
+  var requestResult = this._database._connection.GET(this._prefixurl(
+    `/_api/replication/revisions/tree?collection=${this._name}&verification=true&batchId=${batch.id}`));
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-collection.js:1541]:
```suggestion
    `/_api/replication/revisions/tree?collection=${encodeURIComponent(this._name)}&verification=true&batchId=${batch.id}`));
```

File: js/client/modules/@arangodb/arango-collection.js [hunk 4]
```diff
@@ -1527,3 +1527,30 @@ ArangoCollection.prototype.loadIndexesIntoMemory = function () {
...
+};
+
+//////////////////////////////////////////////////////////////////////////////
+/// @brief MerkleTreeRebuilding
+//////////////////////////////////////////////////////////////////////////////
+
+ArangoCollection.prototype._revisionTreeRebuild = function() {
+  // For some reason we need a batch ID here, which is not used!
+  var requestResult = this._database._connection.POST(this._prefixurl(
+    `/_api/replication/revisions/tree?collection=${this._name}&batchId=42`), {});
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-collection.js:1554]:
```suggestion
    `/_api/replication/revisions/tree?collection=${encodeURIComponent(this._name)}&batchId=42`), {});
```

File: js/client/modules/@arangodb/arango-collection.js [hunk 5]
```diff
@@ -1527,3 +1527,30 @@ ArangoCollection.prototype.loadIndexesIntoMemory = function () {
...
+  return requestResult;
+};
+
+//////////////////////////////////////////////////////////////////////////////
+/// @brief MerkleTreeRebuilding
+//////////////////////////////////////////////////////////////////////////////
+
+ArangoCollection.prototype._revisionTreeRebuild = function() {
+  // For some reason we need a batch ID here, which is not used!
+  var requestResult = this._database._connection.POST(this._prefixurl(
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-collection.js:1553]: `requestResult` is not used here, so we can remove it.
Or, even better, we should check the return value via `arangosh.checkRequestResult(requestResult);`

File: arangod/RocksDBEngine/RocksDBV8Functions.cpp [hunk 6]
```diff
@@ -221,6 +223,86 @@ static void JS_CollectionRevisionTreeCorrupt(v8::FunctionCallbackInfo<v8::Value>
...
+
+  {
+    TRI_vocbase_t& vocbase = collection->vocbase();
+    auto& server = vocbase.server();
+    RocksDBEngine& engine = server.getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+    RocksDBReplicationManager* manager = engine.replicationManager();
+    double ttl = 3600;
+    RocksDBReplicationContext* ctx 
+      = manager->createContext(engine, ttl, SyncerId{17}, ServerId{17}, "");
+    RocksDBReplicationContextGuard guard(manager, ctx);
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBV8Functions.cpp:248]: I think `ctx` can be a nullptr here. We should check the return value and throw an exception if it is a nullptr.

Comment (neunhoef) [reply arangod/RocksDBEngine/RocksDBV8Functions.cpp:248]: Done.

File: js/client/modules/@arangodb/arango-collection.js (hunk 5 above)
Comment (neunhoef) [reply js/client/modules/@arangodb/arango-collection.js]: Done.

**Pull request merged**